### ISSUE TITLE
MIRZAG snakemake version crashes when input mrnas are specified with "|"

### ISSUE THREAD

**Issue by fgypas** [guma44/MIRZAG#8]
Input files example in **Alignment extraction pipeline**:

```
>ENST00000632684.1|ENSG00000282431.1|AC245427.8
GGGACAGGGGGC
>ENST00000434970.2|ENSG00000237235.2|TRDD2
CCTTCCTAC
>ENST00000448914.1|ENSG00000228985.1|TRDD3
ACTGGGGGATACG
>ENST00000415118.1|ENSG00000223997.1|TRDD1
GAAATAGT
>ENST00000631435.1|ENSG00000282253.1|AC239618.6
GGGACAGGGGGC
```

Types of output paths of **Alignment extraction pipeline** (input for mirza-g):

`mirzag/ENST00000634191.1__ENSG00000276125.4__CCL4L2`

Using such files as input on the **MIRZA-G pipeline** and using the following parameter in the config:

`the settings_split_by: '"|"'`

We get the following error in the rule **calculate_mirza**:

```
Traceback (most recent call last):
  File "/usr/bin/rg_calculate_MIRZA.py", line 782, in <module>
    main(options)
  File "/usr/bin/rg_calculate_MIRZA.py", line 124, in main
    multiple_alignment_dict = read_multiple_alignments(options.msa, coords)
  File "/usr/bin/rg_calculate_MIRZA.py", line 726, in read_multiple_alignments
    f = t.extractfile(coord[0])
  File "/usr/local/lib/python2.7/tarfile.py", line 2145, in extractfile
    tarinfo = self.getmember(member)
  File "/usr/local/lib/python2.7/tarfile.py", line 1829, in getmember
    raise KeyError("filename %r not found" % name)
KeyError: "filename 'ENST00000452239.1|ENSG00000088926.13|F11' not found"
```

The problem is that the actual filename is `ENST00000452239.1__ENSG00000088926.13__F11`.

If we change the following parameter and re-run the pipeline:

`the settings_split_by: '"__"'`

We get a different error at the rule **count_miRNA_seeds_and_filter_duplicates** (part of mirzag-seed pipeline):

```
Traceback (most recent call last):
  File "/usr/bin/rg_count_miRNA_seeds_and_filter_duplicates.py", line 183, in <module>
    main(options)
  File "/usr/bin/rg_count_miRNA_seeds_and_filter_duplicates.py", line 152, in main
    data['newid'] = [i.split(options.split_by)[options.index_after_split] for i in data.id]
IndexError: list index out of range
```

I think the simpler solution is to roll back to the version where we used everywhere the "|" separator.

@jsurkont Can we take a look on this issue today or tommorow?

**Comment by jsurkont** [comment]
I can now recall that there was an issue with the CWL implementation. CWL doesn't accept file names containing the pipe character, so we probably need to avoid them.
@fgypas we can discuss the solution later this afternoon

**Comment by fgypas** [comment]
> @fgypas we can discuss the solution later this afternoon

Perfect

**Comment by fgypas** [comment]
Fixed in 136ad38a54291d85ac313e5c6385f4b263ce58ba
